R/borrowing_class.R

```diff
@@ -1,58 +1,18 @@
-# class union ----
-setClassUnion("PriorOrNULL", c("Prior", "NULL"))
-
 #' `Borrowing` Class
 #'
 #' A class for defining borrowing details. Objects of class
-#' `Borrowing` should not be created directly but by the constructor
-#' [borrowing_details()].
+#' `Borrowing` should not be created directly but by the constructors
+#' [hierarchical_commensurate_borrowing(), no_borrowing(), full_borrowing()].
 #'
-#' @slot method character. The type of borrowing to perform. It
-#' must be one of: `'BDB'`, `'Full borrowing'`, or `'No borrowing'`. See `?borrowing_details` for
-#' more information.
-#' @slot ext_flag_col character. The name of the column in
-#' the data matrix that corresponds to the external control flag (`1`/`0` or
-#' `TRUE`/`FALSE`). This identifies a patient as belonging to the external
-#' control cohort.
-#' @slot tau_prior `Prior`. Object of class `Prior` defining the hyperprior on the
-#' "commensurability parameter". See `?borrowing_details` for more information.
 #' @slot data_stan_code string. Code to include in the Stan data program block.
-.borrowing_class <- setClass(
+#' @slot method_name string. The name of the method.
+setClass(
   "Borrowing",
   slots = c(
-    method = "character",
-    ext_flag_col = "character",
```

Comment:
Should we keep `ext_flag_col` in the borrowing class? It seems it would be common to all inherited classes?

Reply:
@gravesti good call, I took it out of full borrowing briefly but it would be too much to work back in